### PR TITLE
Made tgc test data change trigger tgc test in CI

### PR DESCRIPTION
**Pull request by shuyama1**
Currently, changes made solely in the [mmv1/third_party/tgc/tests/data](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/tgc/tests/data) will result in TGC tests being skipped during PR checks. This could lead to false positives in TGC-specific test fix PRs, as seen in https://github.com/GoogleCloudPlatform/magic-modules/pull/11558

**Release Note Template for Downstream PRs (will be copied)**

```release-note:none

```

### CODE REVIEW

File: .github/workflows/test-tgc.yml [hunk 1]
```diff
@@ -67,7 +67,7 @@ jobs:
     - name: Check for Code Changes
       id: pull_request
       run: |
-        gofiles=$(git diff --name-only HEAD~1 | { grep -e "\.go$" -e "go.mod$" -e "go.sum$" || test $? = 1; })
+        gofiles=$(git diff --name-only HEAD~1 | { grep -e "\.go$" -e "go.mod$" -e "go.sum$" -e "tfplan2cai/testdata/templates/" || test $? = 1; })
```
Comment (melinath) [inline .github/workflows/test-tgc.yml:70]: possibly anything in testdata?

```suggestion
        gofiles=$(git diff --name-only HEAD~1 | { grep -e "\.go$" -e "go.mod$" -e "go.sum$" -e "tfplan2cai/testdata/" || test $? = 1; })
```

**Review by melinath — CHANGES_REQUESTED** [review]

**Review by melinath — APPROVED** [review]
meant to approve, whoops!

File: .github/workflows/test-tgc.yml (hunk 1 above)
Comment (shuyama1) [reply .github/workflows/test-tgc.yml:70]: yeah, that makes sense! Thanks for the quick review

**Comment by shuyama1** [comment]
The TGC failure is expected as it's already failed at head. The fix of the failure is in https://github.com/GoogleCloudPlatform/magic-modules/pull/11581

**Pull request merged**